### ISSUE TITLE
Can the synchronicity of "ALTER TABLE … DELETE Statement" queries be enforced programmatically?

### ISSUE THREAD

**Issue by omerkaya1** [ClickHouse/clickhouse-go#312]
Hello.

**TL;DR:** There are two calls to the same table: one that deletes an entry from the table and another that fetches all entries from it, each occurring synchronously, which causes inconsistent data to be returned to the issuer of the second call.

I've searched the documentation and found out the following:

The corresponding [ClickHouse documentation entry](https://clickhouse.tech/docs/en/sql-reference/statements/alter/delete/) on ```ALTER TABLE … DELETE Statement``` states that by default all such query calls are asynchronous.

It also states that the synchronicity can be enforced through a setting option [```mutations_sync```](https://clickhouse.tech/docs/en/operations/settings/settings/#mutations_sync).
However, I haven't been able to find a proper way to do so through the functionality provided by the driver.

So, my question is twofold: is there a way to change the default [mutations](https://clickhouse.tech/docs/en/sql-reference/statements/alter/#mutations) behaviour and if so, how to properly do that using the driver?

### Actual behaviour

- There are 2 entries in a table;
- Delete a record from the table -> no error;
- Read data from the table -> `a` and `b` records both are returned.

### Desired behaviour

- There are 2 entries in a table (`a` and `b`);
- Delete the `a` record from the table -> no error;
- Read data from the table -> only `b` record is returned.

Thank you in advance.
